### DISCUSSION TITLE
DisallowConcurrentExecution with multiple groups

### DISCUSSION THREAD

**Discussion by Ghyath-Serhal** [quartznet/quartznet#1900] [Q&A] [answered]
I have a multitenant application and I have one instance of Quartz scheduler that will handle all the tenants.
Each job will have a different group name based on the tenant.
Same job might run for different tenants at the same time.

I was using the [**DisallowConcurrentExecution**] property, but this will prevent the same job to run for two tenants at the same time.

Currently I removed the [**DisallowConcurrentExecution**] property and I am handling this manually in the **VetoJobExecution** in the **Trigger listener**.

It would be a good feature to add if the **DisallowConcurrentExecution** can work per **group**.

**Comment by lahma** [comment, marked as answer]
`DisallowConcurrentExecution` works based on job key. If you register same job type multiple times with different job key (for each tenant) they should be able to run concurrently.
